packages/medusa/src/services/product.ts

```diff
@@ -0,0 +1,1067 @@
+import { MedusaError } from "medusa-core-utils"
+import { EntityManager } from "typeorm"
+import { SearchService } from "."
+import { TransactionBaseService } from "../interfaces"
+import { PriceSelectionContext } from "../interfaces/price-selection-strategy"
+import { Product, ProductTag, ProductType, ProductVariant } from "../models"
+import { CartRepository } from "../repositories/cart"
+import { ImageRepository } from "../repositories/image"
+import {
+  FindWithoutRelationsOptions,
+  ProductRepository,
+} from "../repositories/product"
+import { ProductOptionRepository } from "../repositories/product-option"
+import { ProductTagRepository } from "../repositories/product-tag"
+import { ProductTypeRepository } from "../repositories/product-type"
+import { ProductVariantRepository } from "../repositories/product-variant"
+import PriceSelectionStrategy from "../strategies/price-selection"
+import { FindConfig, Selector } from "../types/common"
+import {
+  CreateProductInput,
+  FilterableProductProps,
+  ProductOptionInput,
+  UpdateProductInput,
+} from "../types/product"
+import { buildQuery, setMetadata, validateId } from "../utils"
+import { formatException } from "../utils/exception-formatter"
+import { omitRelationIfExists } from "../utils/omit-relation-if-exists"
+import EventBusService from "./event-bus"
+import ProductCollectionService from "./product-collection"
+import ProductVariantService from "./product-variant"
+
+type InjectedDependencies = {
+  manager: EntityManager
+  productOptionRepository: typeof ProductOptionRepository
+  productRepository: typeof ProductRepository
+  productVariantRepository: typeof ProductVariantRepository
+  productTypeRepository: typeof ProductTypeRepository
+  productTagRepository: typeof ProductTagRepository
+  imageRepository: typeof ImageRepository
+  cartRepository: typeof CartRepository
+  productVariantService: ProductVariantService
+  productCollectionService: ProductCollectionService
+  searchService: SearchService
+  eventBusService: EventBusService
+  priceSelectionStrategy: PriceSelectionStrategy
+}
+
+type PriceListLoadConfig = {
```

Comment:
question: Should we move those types in the type directory under the price list file? just to be consistent? only the injectDependencies is closely related to the class and can't be used elsewhere.
What is opinion on that?